tests/test_pipeline/utils.py

```diff
@@ -116,12 +127,51 @@ def assert_pipeline_forecasts_given_ts_with_prediction_intervals(
     else:
         expected_segments = to_forecast_ts.segments
     assert forecast_ts.segments == expected_segments
-    expected_index = pd.date_range(
-        start=to_forecast_ts.index[-1], periods=horizon + 1, freq=to_forecast_ts.freq, name="timestamp"
-    )[1:]
+
+    if ts.freq is None:
+        expected_index = pd.Index(
+            np.arange(to_forecast_ts.index[-1], to_forecast_ts.index[-1] + horizon + 1), name="timestamp"
+        )[1:]
+    else:
+        expected_index = pd.date_range(
+            start=to_forecast_ts.index[-1], periods=horizon + 1, freq=to_forecast_ts.freq, name="timestamp"
+        )[1:]
     pd.testing.assert_index_equal(forecast_ts.index, expected_index)
+
     assert not forecast_df["target"].isna().any()
     assert not forecast_df["target_0.025"].isna().any()
     assert not forecast_df["target_0.975"].isna().any()
 
     return pipeline
+
+
+def assert_pipeline_predicts(
+    pipeline: AbstractPipeline, ts: TSDataset, start_idx: int, end_idx: int
+) -> AbstractPipeline:
+    predict_ts = deepcopy(ts)
+    pipeline.fit(ts)
+
+    start_timestamp = ts.index[start_idx]
+    end_timestamp = ts.index[end_idx]
+    num_points = end_idx - start_idx + 1
+
+    predict_ts = pipeline.predict(ts=predict_ts, start_timestamp=start_timestamp, end_timestamp=end_timestamp)
+    predict_df = predict_ts.to_pandas(flatten=True)
+
+    if ts.has_hierarchy():
+        expected_segments = ts.hierarchical_structure.get_level_segments(predict_ts.current_df_level)
+    else:
+        expected_segments = predict_ts.segments
+    assert predict_ts.segments == expected_segments
+
+    if ts.freq is None:
+        expected_index = pd.Index(np.arange(start_timestamp, end_timestamp + 1), name="timestamp")
+    else:
+        expected_index = pd.date_range(
+            start=start_timestamp, periods=num_points, freq=predict_ts.freq, name="timestamp"
+        )
```

Comment:
Perhaps there should be a separate function for the creation of the index? We use this logic in many places.

Reply:
Yes, I'll add this logic in a separate PR.